.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDatePeriodParser.cs

```diff
@@ -1079,13 +1106,18 @@ private DateTimeResolutionResult MergeTwoTimePoints(string text, DateObject refe
                 er[1].Text = weekPrefix + " " + er[1].Text;
             }
 
+            var dateContext = GetYearContext(er[0].Text, er[1].Text, text);
+
             var pr1 = this.config.DateParser.Parse(er[0], referenceDate);
             var pr2 = this.config.DateParser.Parse(er[1], referenceDate);
             if (pr1.Value == null || pr2.Value == null)
             {
                 return ret;
             }
 
+            pr1 = dateContext.ProcessDateEntityParsingResult(pr1);
```

Comment:
"Process" for what?

Reply:
This is to update the ParsingResult to use the date context (currently only "year" information is applied)